src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeExtensions.cs

```diff
@@ -44,6 +37,11 @@ private static bool MatchesArgumentTypes(this MethodInfo mi, Type[] argTypes)
             Debug.Assert(mi != null);
```

Comment:
Also, if all or almost all of the build-only builds pass and all or almost all of the release build-and-test pass and all the debug build-and-test fail, then an invalid debug-only assertion is always a strong possibility.